integration-tests/src/test/java/org/apache/druid/tests/indexer/ITCompactionTaskTest.java

```diff
@@ -164,18 +211,31 @@ private void loadData(String indexTask) throws Exception
     );
   }
 
-  private void compactData(String compactionResource, GranularityType newSegmentGranularity) throws Exception
+  private void compactData(String compactionResource, GranularityType newSegmentGranularity, GranularityType newQueryGranularity) throws Exception
   {
     String template = getResourceAsString(compactionResource);
     template = StringUtils.replace(template, "%%DATASOURCE%%", fullDatasourceName);
+    // For the new granularitySpec map
+    Map<String, String> granularityMap = new HashMap<>();
+    if (newSegmentGranularity != null) {
+      granularityMap.put("segmentGranularity", newSegmentGranularity.name());
+    }
+    if (newQueryGranularity != null) {
+      granularityMap.put("queryGranularity", newQueryGranularity.name());
+    }
+    template = StringUtils.replace(
+        template,
+        "%%GRANULARITY_SPEC%%",
+        jsonMapper.writeValueAsString(granularityMap)
+    );
```

Comment:
I think this means that there will always be a granulartySpec provided as part of the compaction task spec. Do we have a test running a compact task without specifying a granularitySpec. I'm trying to think of what happens in an upgrade/downgrade scenario where the compaction job being submitted may not have the granularitySpec as part of the compaction spec

Reply:
There are compaction IT (existing ones) still use compaction spec (/indexer/wikipedia_compaction_task.json) without granularitySpec (for example, `testCompaction()`).

Reply:
Basically, 
```
    template = StringUtils.replace(
        template,
        "%%GRANULARITY_SPEC%%",
        jsonMapper.writeValueAsString(granularityMap)
    );
```
will be a no-op for those tests that uses compaction spec without granularitySpec (/indexer/wikipedia_compaction_task.json)